lib/pharos/scripts/configure-kube.sh

```diff
@@ -2,6 +2,22 @@
 
 set -eu
 
+if [ -e /etc/systemd/system/kubelet.service.d/5-pharos-etcd.conf ]; then
+    # remove etcd config because it conflicts with kubeadm
+    rm /etc/systemd/system/kubelet.service.d/5-pharos-etcd.conf
+    systemctl stop kubelet
+    systemctl daemon-reload
+fi
+
+if [ -e /etc/systemd/system/kubelet.service.d/5-pharos-kubelet-proxy.conf ]; then
+    # remove proxy config because it conflicts with kubeadm
+    rm /etc/systemd/system/kubelet.service.d/5-pharos-kubelet-proxy.conf
+    systemctl stop kubelet
+    systemctl daemon-reload
+fi
+
 apt-mark unhold kubelet kubectl kubeadm
-apt-get install -y kubelet=${KUBE_VERSION}-00 kubectl=${KUBE_VERSION}-00 kubeadm=${KUBE_VERSION}-00
-apt-mark hold kubelet kubectl kubeadm
\ No newline at end of file
+apt-get install -y kubelet=${KUBE_VERSION}-00 kubectl=${KUBE_VERSION}-00 kubeadm=${KUBEADM_VERSION}-00
+apt-mark hold kubelet kubectl kubeadm
+
+systemctl restart kubelet
```

Comment:
Same here, this is unconditionally restarting the kubelet?

Reply:
The `apt-get install kubeadm` will already trigger `systemctl restart kubelet` as part of the postinst script during an initial install.

I suppose this restart is only needed if `kubeadm` is already installed, but the `/etc/kubernetes/kubelet.conf` was missing during the previous `configure-kubelet-proxy.sh` / `configure-etcd.sh` phase, so we need to restart the `kubelet` without the temporary dropins?

Maybe this should be part of the `if [ -e /etc/systemd/system/kubelet.service.d/5-pharos-*.conf ]` blocks then?

Reply:
Maybe this should be part of the `if [ -e /etc/systemd/system/kubelet.service.d/5-pharos-*.conf ]` blocks?

Although for the initial install case, that would then end up restarting the `kubelet` in the default configuration before the `apt-get install kubeadm`, which again exposes the unauthenticated kubelet API by default... :unamused: